frontend/app/routes/protected/person-case/last-name.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react';
+
+import { data, redirect, useFetcher, useParams } from 'react-router';
+import type { ActionFunctionArgs, RouteHandle } from 'react-router';
+
+import { useTranslation } from 'react-i18next';
+
+import type { Route } from '../+types';
+
+import { requireAuth } from '~/.server/utils/auth-utils';
+import { lastNameSchema } from '~/.server/validation';
+import { Button } from '~/components/button';
+import { ButtonLink } from '~/components/button-link';
+import { ErrorSummary } from '~/components/error-summary';
+import { InputField } from '~/components/input-field';
+import { PageTitle } from '~/components/page-title';
+import { getFixedT } from '~/i18n-config.server';
+import { i18nRoutes } from '~/i18n-routes';
+import { handle as parentHandle } from '~/routes/protected/layout';
+import { getPathByFile } from '~/utils/route-utils';
+import { getErrorList } from '~/utils/zod-utils';
+
+export const handle = {
+  i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
+} as const satisfies RouteHandle;
+
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+  return { documentTitle: t('protected:person-case.page-title') };
+}
+
+export function meta({ data }: Route.MetaArgs) {
+  return [{ title: data.documentTitle }];
+}
+
+export async function action({ context, request }: ActionFunctionArgs) {
+  const formData = await request.formData();
+  const t = await getFixedT(request, handle.i18nNamespace);
+  const schema = lastNameSchema({
+    errorMessages: {
+      format_error: t('protected:person-case.last-name.format-error'),
+      invalid_type_error: t('protected:person-case.last-name.invalid-type-error'),
+      max_length_error: t('protected:person-case.last-name.max-length-error'),
+      required_error: t('protected:person-case.last-name.required-error'),
+    },
+  });
+  const parsedDataResult = schema.safeParse(String(formData.get('last-name') ?? ''));
+
+  if (!parsedDataResult.success) {
+    return data({ errors: getErrorList(parsedDataResult.error.flatten()) }, { status: 400 });
+  }
+
+  return redirect(getPathByFile('routes/protected/request.tsx', i18nRoutes, request));
+}
+
+export default function LastName() {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const params = useParams();
+  const fetcher = useFetcher<typeof action>();
```

Comment:
I *think* the idiomatic way to do this in React Router 7 is this:

``` typescript
import type { Info } from './+types/last-name';
const fetcher = useFetcher<Info['actionData']>();
```

Reply:
Read my comment for line 72 before considering this.

Reply:
`actionData` is available in `Route.ComponentProps`, [see](https://github.com/DTS-STN/future-sir/pull/115/files#r1927478765)